src/com/sap/piper/TemporaryCredentialsUtils.groovy

```diff
@@ -8,47 +8,63 @@ class TemporaryCredentialsUtils implements Serializable {
             if (useCredentials) {
-                writeCredentials(credentialItems, credentialsDirectory, credentialsFileName)
+                writeCredentials(credentialItems, credentialsDirectories, credentialsFileName)
             }
             body()
         }
         finally {
             if (useCredentials) {
-                deleteCredentials(credentialsDirectory, credentialsFileName)
+                deleteCredentials(credentialsDirectories, credentialsFileName)
             }
         }
     }
 
-    private void writeCredentials(List credentialItems, String credentialsDirectory, String credentialsFileName) {
+    private void writeCredentials(List credentialItems, List credentialsDirectories, String credentialsFileName) {
         if (!credentialItems) {
             script.echo "Not writing any credentials."
             return
         }
 
-        assertSystemsFileExists(credentialsDirectory)
+        Boolean systemsFileFound = false
+        for (int i = 0; i < credentialsDirectories.size(); i++) {
+            if (credentialsDirectories[i] && !credentialsDirectories[i].endsWith("/")) {
+                credentialsDirectories[i] += '/'
+            }
+            if (script.fileExists("${credentialsDirectories[i]}systems.yml") || script.fileExists("${credentialsDirectories[i]}systems.yaml") || script.fileExists("${credentialsDirectories[i]}systems.json")) {
+                String credentialJson = returnCredentialsAsJSON(credentialItems)
 
-        String credentialJson = returnCredentialsAsJSON(credentialItems)
+                script.echo "Writing credentials file with ${credentialItems.size()} items to ${credentialsDirectories[i]}."
+                script.writeFile file: credentialsDirectories[i] + credentialsFileName, text: credentialJson
 
-        script.echo "Writing credentials file with ${credentialItems.size()} items."
-        script.dir(credentialsDirectory) {
-            script.writeFile file: credentialsFileName, text: credentialJson
+                systemsFileFound = true
+            }
+        }
+        if (!systemsFileFound) {
+            script.error("None of the directories ${credentialsDirectories} contains any of the files systems.yml, systems.yaml or systems.json. " +
+                "One of those files is required in order to activate the integration test credentials configured in the pipeline configuration file of this project. " +
+                "Please add the file as explained in the SAP Cloud SDK documentation.")
         }
     }
 
-    private void deleteCredentials(String credentialsDirectory, String credentialsFileName) {
-        script.echo "Deleting credentials file."
-        script.dir(credentialsDirectory) {
-            script.sh "rm -f ${credentialsFileName}"
+    private void deleteCredentials(List credentialsDirectories, String credentialsFileName) {
+        for (int i = 0; i < credentialsDirectories.size(); i++) {
+            if(!credentialsDirectories[i].endsWith('/'))
```

Comment:
The check is inconsistent with line 40. Can the list have null entries?

Reply:
True I missed it. I'll add the check as suggested above.